### PR TITLE
fix(resolver): fixes a bug where resolved dependent subscriptions don't

### PR DESCRIPTION
**Pull request by ecordell**
detect updates to catalogs

this occurred because we only check for updates when we have information
about the currently installed CSV on the subscription status, which
was omitted from dependent catalogs.

the resolver now records the csv resolved into the subscription spec,
and it is used to populate the installed csv status when available.

### CODE REVIEW

File: pkg/controller/operators/olm/operator.go [hunk 1]
```diff
@@ -1093,6 +1112,10 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 		}
 	case v1alpha1.CSVPhaseDeleting:
 		var immediate int64 = 0
+
+		if err := a.csvQueueSet.Remove(out.GetNamespace(), out.GetName()); err != nil {
```
Comment (njhale) [inline pkg/controller/operators/olm/operator.go]: The name and namespace parameters are reversed here.

Comment (njhale) [reply pkg/controller/operators/olm/operator.go]: Doesn't the following delete step result in the key being enqueued again?

Comment (ecordell) [reply pkg/controller/operators/olm/operator.go]: That explains some logs I saw :) If this is the only thing I will follow up with that fix.

Comment (ecordell) [reply pkg/controller/operators/olm/operator.go]: I have also fixed in #841

**Pull request merged**